tests/integration/integration_test.go

```diff
@@ -1146,10 +1146,76 @@ func TestStackReferencePython(t *testing.T) {
 		Config: map[string]string{
 			"org": os.Getenv("PULUMI_TEST_OWNER"),
 		},
+		EditDirs: []integration.EditDir{
+			{
+				Dir:      "step1",
+				Additive: true,
+			},
+			{
+				Dir:      "step2",
+				Additive: true,
+			},
+		},
 	}
 	integration.ProgramTest(t, opts)
 }
 
+func TestMultiStackReferencePython(t *testing.T) {
+	if runtime.GOOS == WindowsOS {
+		t.Skip("Temporarily skipping test on Windows - pulumi/pulumi#3811")
+	}
+	if owner := os.Getenv("PULUMI_TEST_OWNER"); owner == "" {
+		t.Skipf("Skipping: PULUMI_TEST_OWNER is not set")
+	}
+
+	// build a stack with an export
+	exporterOpts := &integration.ProgramTestOptions{
+		Dir: filepath.Join("stack_reference_multi", "python", "exporter"),
+		Dependencies: []string{
+			filepath.Join("..", "..", "sdk", "python", "env", "src"),
+		},
+		Quick: true,
+		Config: map[string]string{
+			"org": os.Getenv("PULUMI_TEST_OWNER"),
+		},
+		NoParallel:           true,
+		SkipLifeCycleDestroy: true, // do this at the end after the importer finishes
+	}
+
+	// we're going to manually initialize and then defer the deletion of this stack
+	exporterPt := integration.ProgramTestManualLifeCycle(t, exporterOpts)
+	exporterPt.TestFinished = false
+	err := exporterPt.TestLifeCyclePrepare()
+	assert.NoError(t, err)
+	err = exporterPt.TestLifeCycleInitialize()
+	assert.NoError(t, err)
+	err = exporterPt.TestPreviewUpdateAndEdits()
+	assert.NoError(t, err)
+
+	defer func() {
```

Comment:
nit: this defer statement should happen before `TestPreviewUpdateAndEdits()` so that cleanup happens if the test case panics. This would be consistent with the order of operations within the old `ProgramTest()` method.

Reply:
Ah yes, thanks for pointing that out!